src/west/manifest.py

```diff
@@ -41,7 +45,9 @@ def default_path():
     '''Return the path to the default manifest in the west directory.
 
     Raises WestNotFound if called from outside of a west working directory.'''
-    return os.path.join(util.west_dir(), 'manifest', 'manifest.yml')
+    return os.path.join(util.west_topdir(),
+                        config.get('self', 'path', fallback='manifest'),
```

Comment:
This doesn't seem right. There is no default or fallback: this value simply must be present, or the config file has gotten corrupted and we've lost information needed to track the installation. We can't guess this value since there may be multiple projects with a manifest file inside.

Reply:
yep. This was actually something I wanted to raise when I was doing it.
(and then forgot about it again :( )
So thanks for raising this.

`fallback='manifest'` was to have similar behavior as before and it is was required to ensure the tests are passing in this commit (deferred `init` is added in following commit)
(Personally, I prefer passing tests in each commit)

So question is, do we want to default to `zephyr` which will be the default project with manifest in future, or abort with an error ?

Reply:
I suggest we bail out with a clear error message that instructs the user to start anew

Reply:
As discussed elsewhere, solution for now is to remove the defaults and no specific error handling (meaning default python stack trace if key is missing)

Reply:
Why not at least catch the error if the information is missing and call `log.die()` with a useful message?

Reply:
done